crates/core/src/table/table_parquet_options.rs

```diff
@@ -23,6 +32,34 @@ pub fn build_writer_properties(
     })
 }
 
+#[cfg(feature = "datafusion")]
+pub fn build_writer_properties_factory(
```

Comment:
Do we want this to handle the case where an encryption factory id is set in the `TableParquetOptions` somehow? Or maybe the `TableEncryption`  type could be integrated in the table properties when those are refactored and then used to create a `KMSWriterPropertiesFactory`? I guess it's not yet clear to me how everything will fit together end-to-end when using KMS encryption.

Reply:
Yeah, as mentioned, I think the idea is to create a DeltaTableOptions trait. I would envision it as having two methods:
1. Generate datafusion `TableOptions`. (for reading) - we may want to split this more.
2. Generate a `WriterPropertiesFactory`.

#2 is where it could generate a `KMSWriterPropertiesFactory`. I plan to have a go at adding this over the next day and then you can see if that is enough to generate a workable KMS or if something is missing.